### ISSUE TITLE
Numba 0.57 breaks fmpy

### ISSUE THREAD

**Issue by carlfischerjba** [OasisLMF/OasisLMF#1267]
## Issue Description

numba 0.57 was released a few days ago and causes deprecation warnings to be reported by fmpy. These appear in `stderror.err` and cause the ktools monitor to cancel the analysis.

## Version / Environment information

- `oasislmf==1.27.2`

## Example data / logs

```
$ cat log/stderror.err
/home/.../.../farmcat_venv/lib/python3.8/site-packages/oasislmf/pytools/fm/stream_sparse.py:379: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @jit(cache=True)
...
```

**Comment by sambles** [comment]
There is a related PR here to redirect warning messages over to a log file, but its not been pushed in stable. 
https://github.com/OasisLMF/OasisLMF/pull/1229

I'll test it using numba 0.57, and backport the fix over to `1.27.3` (due next week) if its working

**Comment by sambles** [comment]
Keeping issue open for a longer term fix. It should be possible to redirect the warnings from numba to the logging module using `logging.captureWarnings(capture)`
